python/src/scipp/plotting/model.py

```diff
@@ -86,14 +86,14 @@ def __init__(self,
                 else:
                     coord_list[dim] = to_bin_edges(coord, dim)
 
-            # Create a PlotArray helper object that supports slicing where new
-            # bin-edge coordinates can be attached to the data
-            self.data_arrays[name] = PlotArray(data=array.data,
-                                               meta=coord_list)
+            self.backup.update({name: {"array": array, "coords": coord_list}})
 
-            # Include masks
-            for m in array.masks:
-                self.data_arrays[name].masks[m] = array.masks[m]
+        # Save a copy of the name for simpler access
+        # Note this needs to be done before calling update_data_arrays
+        self.name = name
```

Comment:
The name of the last item?

Reply:
Yes, this has always been the case.
For 1d plots, there can be more than one item in in the dict, but `self.name` is not used.
For 2+d plots, there can only be one name.
I've always thought it was not a great way of setting the name, but I didn't find a better way.